### DISCUSSION TITLE
Microservices - Authentication

### DISCUSSION THREAD

**Discussion by rajesh-ryse** [jhipster/generator-jhipster-quarkus#361] [Q&A]
Hello,

Every microservice that I create via jHipster, has code for user authentication using email & password.

Is this really needed for each microservice? 
We rely on central authentication before a request hits any microservice. The service should just deal with the authentication token for roles.

Please advise.

Thanks!

**Comment by mraible** [comment]
If you use OAuth, you can use an external identity provider.

**Comment by rajesh-ryse** [comment]
why a microservice need OAuth or any authentication logic, when a user is authenticated before its request hits the microservice?
